app/models/host/managed.rb

```diff
@@ -485,6 +485,11 @@ def host_inherited_params_objects
     params
   end
 
+  def host_params_objects
+    # Host parameters should always be first for the uniq order
+    (host_parameters + host_inherited_params_objects).uniq {|param| param.name}
```

Comment:
I think this also needs to reverse `host_inherited_params_objects`.  If uniq's working from most to least specific, then a CommonParameter looks like it'll override an ancestor parameter.  The UI is showing it correctly but the API seems to work in the wrong order.